modules/020-cni-cilium/templates/agent/rbac-for-us.yaml

```diff
@@ -0,0 +1,121 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: agent
+  namespace: d8-{{ .Chart.Name }}
+{{ include "helm_lib_module_labels" (list . (dict "app" "agent")) | indent 2 }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: d8:cni-cilium:agent
+{{ include "helm_lib_module_labels" (list . (dict "app" "agent")) | indent 2 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: d8:cni-cilium:agent
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: d8-{{ .Chart.Name }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: d8:cni-cilium:agent
+{{ include "helm_lib_module_labels" (list . (dict "app" "agent")) | indent 2 }}
+rules:
+- apiGroups:
+  - networking.k8s.io
+  resources:
+  - networkpolicies
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - discovery.k8s.io
+  resources:
+  - endpointslices
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  - services
+  - nodes
+  - endpoints
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - pods/finalizers
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+  - delete
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  - nodes/status
+  verbs:
+  - patch
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - secrets
```

Comment:
This is a cluster role that allows reading secrets. I do not understand why the cilium access rights have to be so powerful.

Reply:
L7 proxy configuration may reference secrets. We haven't tested the L7 functionality of Cilium in any way. But I'd like to keep this rules. Just in case. No harm, right?

Reply:
>why the cilium access rights have to be so powerful

Aren't we trusting Cilium with everything already? It can literally inspect every single by of every single packet that passes around the cluster.

Reply:
You can have no access to the cilium binary and its process memory but easily access the pod filesystem and get the token to read all secrets. The fewer holes you have, the less the surface of attack will be.